### PR TITLE
Supoort mgmt readme generation, Remove karma generation for modular and Add tshy in .gitignore

### PR DESCRIPTION
**Pull request by kazrael2119**
fixes https://github.com/Azure/autorest.typescript/issues/2637
fixes https://github.com/Azure/autorest.typescript/issues/2656
fixes https://github.com/Azure/autorest.typescript/issues/2657

### CODE REVIEW

File: packages/typespec-test/test/anomalyDetector/generated/typespec-ts/README.md [hunk 1]
```diff
@@ -17,55 +19,34 @@ a kind of group based detection, this detection will find inconsistency ones in
...
-### Logging
-
-Enabling logging may help uncover useful information about failures. In order to see a log of HTTP requests and responses, set the `AZURE_LOG_LEVEL` environment variable to `info`. Alternatively, logging can be enabled at runtime by calling `setLogLevel` in the `@azure/logger`:
-
-```javascript
-const { setLogLevel } = require("@azure/logger");
-
-setLogLevel("info");
-```
+`AnomalyDetectorClient` is the primary interface for developers using the AnomalyDetector REST client library. Explore the methods on this client object to understand the different features of the AnomalyDetector service that you can access.
```
Comment (qiaozha) [inline packages/typespec-test/test/anomalyDetector/generated/typespec-ts/README.md:51]: why there's no other parts after `Key concepts` in this case?

File: packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/README.md [hunk 2]
```diff
@@ -1,56 +1,37 @@
...
-Azure Messaging EventGrid Client
-
-**Please rely heavily on our [REST client docs](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/rest-clients.md) to use this library**
+This package contains an isomorphic SDK (runs both in Node.js and in browsers) for Azure.Messaging.EventGrid REST client.
 
-Key links:
+Azure Messaging EventGrid Client
 
-- [Package (NPM)](https://www.npmjs.com/package/@msinternal/eventgrid)
+[Package (NPM)](https://www.npmjs.com/package/@msinternal/eventgrid) |
```
Comment (qiaozha) [inline packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/README.md:7]: where are the other key links? I remember there's four of them?

File: packages/typespec-test/test/hierarchy_generic/generated/typespec-ts/README.md [hunk 3]
```diff
@@ -1,56 +1,37 @@
...
-### Logging
-
-Enabling logging may help uncover useful information about failures. In order to see a log of HTTP requests and responses, set the `AZURE_LOG_LEVEL` environment variable to `info`. Alternatively, logging can be enabled at runtime by calling `setLogLevel` in the `@azure/logger`:
-
-```javascript
-const { setLogLevel } = require("@azure/logger");
-
-setLogLevel("info");
-```
+`FooClient` is the primary interface for developers using the Foo REST client library. Explore the methods on this client object to understand the different features of the Foo service that you can access.
```
Comment (qiaozha) [inline packages/typespec-test/test/hierarchy_generic/generated/typespec-ts/README.md:36]: this one also have no trouble shooting section and contributing section.

File: packages/typespec-test/test/anomalyDetector/generated/typespec-ts/README.md (hunk 1 above)
Comment (kazrael2119) [reply packages/typespec-test/test/anomalyDetector/generated/typespec-ts/README.md:51]: the package scope name is not azure or azure-rest

File: packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/README.md (hunk 2 above)
Comment (kazrael2119) [reply packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/README.md:7]: in main, it only has one link https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/README.md

File: packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/package.json [hunk 4]
```diff
@@ -45,7 +45,7 @@
     "@azure/core-rest-pipeline": "^1.5.0",
     "@azure/logger": "^1.0.0",
     "tslib": "^2.6.2",
-    "@azure/core-lro": "3.0.0",
+    "@azure/core-lro": "^3.0.0",
```
Comment (qiaozha) [inline packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/package.json:48]: Is it correct for autorest generated rlc to use core-lro v3 as a dependency? @MaryGao

**Review by qiaozha — APPROVED** [review]

File: packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/package.json (hunk 4 above)
Comment (MaryGao) [reply packages/autorest.typescript/test/rlcIntegration/generated/dpgCustomization/package.json:48]: yes, this is correct.
- RLC from swagger/typespec: v3
- HLC from swagger: v2
- Modular from typespec: v3

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/README.md [hunk 5]
```diff
@@ -1,46 +1,76 @@
...
-- LTS versions of Node.js
+- [LTS versions of Node.js](https://github.com/nodejs/release#release-schedule)
+- Latest versions of Safari, Chrome, Edge and Firefox.
+
+See our [support policy](https://github.com/Azure/azure-sdk-for-js/blob/main/SUPPORT.md) for more details.
 
 ### Prerequisites
 
-- You must have an [Azure subscription](https://azure.microsoft.com/free/) to use this package.
+- An [Azure subscription][azure_sub].
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/README.md:21]: is this excpected?

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/README.md [hunk 6]
```diff
@@ -1,46 +1,76 @@
...
 npm install @azure/arm-networkanalytics
 ```
 
 ### Create and authenticate a `NetworkAnalyticsClient`
 
-To use an [Azure Active Directory (AAD) token credential](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/samples/AzureIdentityExamples.md#authenticating-with-a-pre-fetched-access-token),
-provide an instance of the desired credential type obtained from the
-[@azure/identity](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#credentials) library.
+To create a client object to access the Azure NetworkAnalytics API, you will need the `endpoint` of your Azure NetworkAnalytics resource and a `credential`. The Azure NetworkAnalytics client can use Azure Active Directory credentials to authenticate.
+You can find the endpoint for your Azure NetworkAnalytics resource in the [Azure Portal][azure_portal].
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/README.md:34]: is this expected: azure_portal?

**Review by MaryGao — COMMENTED** [review]
LGTM

**Pull request merged**